src/main/java/enmasse/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -473,4 +388,244 @@ private ErrorCondition checkFilters(Object partition, Object offset) {
 		
 		return condition;
 	}
+	
+	/**
+	 * Subscribe to the topic
+	 */
+	private void subscribe() {
+		if (partition != null) {
+			// read from a specified partition
+			LOG.debug("Assigning to partition {}", partition);
+			this.consumer.partitionsFor(kafkaTopic, this::partitionsForHandler);
+		} else {
+			LOG.info("No explicit partition for consuming from topic {} (will be automatically assigned)", 
+					kafkaTopic);
+			automaticPartitionAssignment();
+		}
+	}
+	
+	void partitionsForHandler(AsyncResult<List<PartitionInfo>> partitionsResult) {
+		if (partitionsResult.failed()) {
+			LOG.error("Error getting partition info {}", kafkaTopic, partitionsResult.cause());
+			String message = partitionsResult.cause().getMessage();
+			ErrorCondition condition =
+					new ErrorCondition(Symbol.getSymbol(Bridge.AMQP_ERROR_KAFKA_SUBSCRIBE),
+							"Error getting partition info" + (message != null ? ": " + message : ""));
+			sendProtonError(condition);
+			return;
+		}
+		LOG.debug("Getting partitions for " + kafkaTopic);
+		List<PartitionInfo> availablePartitions = partitionsResult.result();
+		Optional<PartitionInfo> requestedPartitionInfo = availablePartitions.stream().filter(p -> p.getPartition() == this.partition).findFirst();
+		
+		if (requestedPartitionInfo.isPresent()) {
+			LOG.debug("Requested partition {} present", partition);
+			if (this.offset != null) {
+				// Don't consume messages until we've called seek()
+				this.consumer.pause();
```

Comment:
With latest modification on the Vert.x Kafka client, can we avoid pausing before assigning ans seeking right ?